### PR TITLE
Bugfix/error manager timing

### PR DESCRIPTION
**Pull request by NicolaiFransen**
Divided errorManager into high and low priority to make everything super efficient

### CODE REVIEW

File: Software/Development/SemesterProject/App/errorManager.c [hunk 1]
```diff
@@ -49,19 +49,38 @@ static errorStatus systemErrorStatus = NO_ERROR;
...
 
     if (errorStatusList.currentErrorStatus == ERROR_HAS_HAPPENED)
     {
         performSafetyReactions();
         updateCurrentErrorStatus();
     }
+}
+
+void performHighPriorityErrorMonitoring(void)
+{
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/App/errorManager.c:71]: Why no need to wait for filter to settle?

File: Software/Development/SemesterProject/App/highPriorityTasks.c [hunk 2]
```diff
@@ -26,10 +18,26 @@ __interrupt void adc_isr(void)
 {
     readHighPrioritySignals();
 
-    AdcRegs.ADCINTFLGCLR.bit.ADCINT1 = 1;
-    PieCtrlRegs.PIEACK.all = PIEACK_GROUP1;   // Acknowledge interrupt to PIE
+    acknowledgeAdcInterrupt();
 
     executeHighPriorityTasks();
```
Comment (AitorTeran) [inline Software/Development/SemesterProject/App/highPriorityTasks.c:24]: Im not 100% percent convinced about readHighPrioritySignals not being part of executeHighPriorityTasks... Seems like read signals should be at the same level as read position or perform control. No big deal anyway

**Review by AitorTeran — APPROVED** [review]
Looking sweet!
Have you checked how long it takes now to do the high priority tasks?
Ship it!¡¿

**Comment by NicolaiFransen** [comment]
The high priority errorMonitoring is reduced to 6us.
When this is merged i will do some more tests on the timing and the UART

File: Software/Development/SemesterProject/App/highPriorityTasks.c (hunk 2 above)
Comment (NicolaiFransen) [reply Software/Development/SemesterProject/App/highPriorityTasks.c:24]: The idea was to be able to leave the interrupt after reading the signals, if a new interrupt had been received. And that is why the acknowledge is in between them

File: Software/Development/SemesterProject/App/errorManager.c (hunk 1 above)
Comment (NicolaiFransen) [reply Software/Development/SemesterProject/App/errorManager.c:71]: If we dont wait for the filters to settle during startup, we will generate some wrong errors.

**Review by murgui — APPROVED** [review]

**Pull request merged**